src/test/FreeC/Backend/Agda/Converter/FuncDeclTests.hs

```diff
@@ -0,0 +1,90 @@
+module FreeC.Backend.Agda.Converter.FuncDeclTests
+  ( testConvertFuncDecls
+  )
+where
+
+import           Test.Hspec
+
+import           FreeC.Backend.Agda.Converter.FuncDecl
+                                                ( convertFuncDecls )
+import           FreeC.Backend.Agda.Pretty      ( )
+import           FreeC.IR.DependencyGraph
+import           FreeC.Monad.Class.Testable
+import           FreeC.Monad.Converter
+import           FreeC.Test.Parser
+import           FreeC.Test.Environment
+import           FreeC.Test.Expectations
+import           FreeC.Pretty                   ( showPretty )
+
+-- | Parses the given IR function declaration, converts it to Agda using
+--   'convertFuncDecls' and sets the expectation that the resulting
+--   Agda code equals the given expected output modulo white space.
+shouldConvertTypeDeclsTo
+  :: DependencyComponent String -> [String] -> Converter Expectation
+shouldConvertTypeDeclsTo inputStrs expectedOutput = do
+  input  <- parseTestComponent inputStrs
+  output <- convertFuncDecls input
+  return (output `prettyShouldBe` showPretty expectedOutput)
+
+-- | Test group for @convertFuncDecls@ tests.
+testConvertFuncDecls :: Spec
+testConvertFuncDecls =
+  describe "FreeC.Backend.Agda.Converter.TypeDecl.convertFuncDecl" $ do
+    it "translates 0-ary functions correctly" $ shouldSucceedWith $ do
+      "Integer" <- defineTestTypeCon "Integer" 0 []
+      "foo"     <- defineTestFunc "foo" 0 "Integer"
+      shouldConvertTypeDeclsTo
+        (NonRecursive "foo :: Integer = 42")
+        ["foo : \x2200 {Shape} {Pos} \x2192 Free Shape Pos (Integer Shape Pos)"]
```

Comment:
The escaped Unicode characters make it very difficult to understand the tests. Since we are using Unicode characters in comments already I'd suggest to use the Unicode character in strings, too. If that's not possible for some reason, I think that there should be a table that shows which escape sequences correspond to which characters in a comment in this file. You should also avoid to mix hexadecimal escapes (e.g. `\x2192`) and decimal escapes (e.g., `\8594`).